### PR TITLE
feat(api-client): added virtual text component and implement it for big responses

### PR DESCRIPTION
**Pull request by amritk**
We need a way to handle giant responses. Responses so big we can't even show them in plain text. So I added this virtual scroller.

Whats left (for a follow up PR):
- we need to play with the threshold to get it right
- also the scrollbar/height calculations don't quite work

test with `https://raw.githubusercontent.com/json-iterator/test-data/refs/heads/master/large-file.json`

Before:
browser crash
After:
image

### CODE REVIEW

File: packages/api-client/src/views/Request/ResponseSection/ResponseBodyVirtual.vue [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+const textContent = computed(() => formatJsonOrYamlString(props.content || ''))
+</script>
+
+<template>
+  <ViewLayoutCollapse class="response-body-virtual">
+    <template #title>Body</template>
+
+    <div
+      class="py-1.5 px-2.5 font-code text-xxs border-1/2 rounded-t border-b-0">
+      Response is too large for syntax highlighting
```
Comment (amritk) [inline packages/api-client/src/views/Request/ResponseSection/ResponseBodyVirtual.vue:20]: Can change this text

File: packages/api-client/src/views/Request/ResponseSection/ResponseSection.vue [hunk 2]
```diff
@@ -55,8 +56,15 @@ const responseCookies = computed(
 
 const sections = ['All', 'Cookies', 'Headers', 'Body']
 type ActiveSections = (typeof sections)[number]
-
 const activeSection = ref<ActiveSections>('All')
+
+/** Threshold for virtualizing text responses in characters */
+const VIRTUALIZATION_THRESHOLD = 5_000_000
```
Comment (amritk) [inline packages/api-client/src/views/Request/ResponseSection/ResponseSection.vue:62]: just a guestimate, could use fine tuning

File: packages/oas-utils/src/helpers/parse.ts [hunk 3]
```diff
@@ -68,7 +68,7 @@ export const transformToJson = (value: string) => {
 export function formatJsonOrYamlString(value: string) {
   // If we don't start with a bracket assume yaml
   const trimmed = value.trim()
-  if (trimmed[0] !== '{') return value
+  if (trimmed[0] !== '{' && trimmed[0] !== '[') return value
```
Comment (amritk) [inline packages/oas-utils/src/helpers/parse.ts:71]: should be safe

File: packages/api-client/src/views/Request/ResponseSection/ResponseBodyVirtual.vue [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+const textContent = computed(() => formatJsonOrYamlString(props.content))
+</script>
+
+<template>
+  <ViewLayoutCollapse class="response-body-virtual">
+    <template #title>Body</template>
+
+    <div
+      class="py-1.5 px-2.5 font-code text-xxs border-1/2 rounded-t border-b-0">
+      Response is too large for syntax highlighting
```
Comment (hanspagel) [inline packages/api-client/src/views/Request/ResponseSection/ResponseBodyVirtual.vue:20]:
```suggestion
      This response body is massive! Syntax highlighting won’t work here.
```

**Review by hanspagel — APPROVED** [review]
What a great PR!!

**Review by marclave — APPROVED** [review]

**Pull request merged**